src/bp/nlu-core/entities/duckling-extractor/index.ts

```diff
@@ -38,55 +27,33 @@ export const DUCKLING_ENTITIES: DucklingDimension[] = [
   'volume'
 ]
 
-const CACHE_PATH = path.join(process.APP_DATA_PATH || '', 'cache', 'sys_entities.json')
-
 // Further improvements:
 // 1- in _extractBatch, shift results ==> don't walk whole array n times (nlog(n) vs n2)
 
 export class DucklingEntityExtractor implements SystemEntityExtractor {
-  public static enabled: boolean
+  public enabled: boolean
   private _provider: DucklingClient
-
-  private static _cache: lru<string, EntityExtractionResult[]>
-  private _cacheDumpEnabled = true
+  private _cache: SystemEntityCacheManager
 
   constructor(private readonly logger?: NLU.Logger) {
+    this.enabled = false
+    this.logger = logger
     this._provider = new DucklingClient(logger)
+    this._cache = new SystemEntityCacheManager(
```

Comment:
this is fine, but if passed it at ctor instead, you could mock it and write unit tests that ensures cache is called correctly

Reply:
Agreed for the sake of testing it will be passed to ctor